### PR TITLE
Add flake8 and mypy to Travis

### PR DESCRIPTION
**Pull request by wsanchez**
Travis isn't running `flake8` or `mypy`.  That seems a bit silly.

### CODE REVIEW

**Comment by wsanchez** [comment]
Waiting on #82 before proceeding.

**Comment by wsanchez** [comment]
Should probably also add `docs`, after #85

**Comment by wsanchez** [comment]
@glyph `flake8` is catching an unused variable in a new test you added.  Can you check on whether that's meant to be used to removed?

**Comment by wsanchez** [comment]
@glyph: `attempted_rooted_replacement`

File: src/hyperlink/test/test_url.py [hunk 1]
```diff
@@ -1112,7 +1112,6 @@ def test_autorooted(self):
 
         attempt_unrooted_absolute = URL(host="foo", path=['bar'], rooted=False)
         normal_absolute = URL(host="foo", path=["bar"])
-        attempted_rooted_replacement = normal_absolute.replace(rooted=True)
```
Comment (wsanchez) [inline src/hyperlink/test/test_url.py]: @glyph: here

Comment (glyph) [reply src/hyperlink/test/test_url.py]: weird. that's a real bug. we should be asserting something about that object

Comment (wsanchez) [reply src/hyperlink/test/test_url.py]: I think that asserting something about that object is unrelated to this PR, though, and is an example of why landing this PR would be good :-)

Comment (wsanchez) [reply src/hyperlink/test/test_url.py]: Not sure how you want to address that.  Can you open a PR with a fix or an issue to track it?

Comment (wsanchez) [reply src/hyperlink/test/test_url.py]: For reference, this came via #91

Comment (glyph) [reply src/hyperlink/test/test_url.py]: Yeah, I guess what I meant was *I* should have asserted something about that object :).

You can feel free to just clean it up here for the time being, and assign any resulting issue to me

Comment (wsanchez) [reply src/hyperlink/test/test_url.py]: Issue #99 for replacing this.

**Pull request merged**